### PR TITLE
rc/8.0.0

### PR DESCRIPTION
**Pull request by martinzigrai**

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -38,12 +38,16 @@ Future<void> _initializeTalsec() async {
       packageName: 'com.aheaditec.freeraspExample',
       signingCertHashes: ['AKoRuyLMM91E7lX/Zqp3u4jMmd0A7hH/Iqozu0TMVd0='],
       supportedStores: ['com.sec.android.app.samsungapps'],
-      malwareConfig: MalwareConfig(
-        blacklistedPackageNames: ['com.aheaditec.freeraspExample'],
-        suspiciousPermissions: [
+      suspiciousAppDetectionConfig: SuspiciousAppDetectionConfig(
```
Comment (tompsota) [inline example/lib/main.dart]: This is a breaking change - we need to raise major in this case

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,50 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [7.6.0] - 2026-05-07
+
+- Android SDK version: 18.3.0
+- iOS SDK version: 6.14.4
+
+### Breaking
```
Comment (tompsota) [inline CHANGELOG.md]: We need to raise major because of this (on all platforms)

File: lib/src/models/android_config.dart [hunk 3]
```diff
@@ -12,7 +12,8 @@ class AndroidConfig {
     required this.packageName,
     required this.signingCertHashes,
     this.supportedStores = const [],
-    this.malwareConfig,
+    @Deprecated('Use suspiciousAppDetectionConfig instead') this.malwareConfig,
```
Comment (tompsota) [inline lib/src/models/android_config.dart]: Keeping both APIs is going to be huge pain. What if I specify both in config? which is going to be used?

Let's remove the old API altogether (on all platforms)

We are doing new major here so we can do this

File: lib/src/models/android_config.g.dart
Comment (tompsota) [inline lib/src/models/android_config.g.dart:1]: Did you generate this using pigeon? Or are those manual edits?

File: android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt [hunk 4]
```diff
@@ -83,3 +88,42 @@ internal fun ExternalIdResult.resolve(result: MethodChannel.Result) {
...
     }
 }
+
+internal fun JSONObject.toMalwareScanScope(): MalwareScanScope {
+    val scopeTypeStr = optString("scanScope", "SIDELOADED_ONLY")
+    val scanScope = runCatching { ScopeType.valueOf(scopeTypeStr) }.getOrDefault(ScopeType.SIDELOADED_ONLY)
+    val trustedInstallSources = optJSONArray("trustedInstallSources")?.let { arr ->
+        (0 until arr.length()).map { arr.getString(it) }
+    }
+    return MalwareScanScope(scanScope, trustedInstallSources)
```
Comment (tompsota) [inline android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt]: SIDELOADED_ONLY is hardcoded as a string in optString, then again as an enum in getOrDefault, then again as a fallback object — three places, two languages, no compile-time guarantee they stay in sync.

The Dart side makes scanScope required (so it can't be missing on the wire) but the Kotlin side still defends against it being absent with a string default. We need only one source of truth.


Suggested approach: just throw exception here if scope is missing.

File: android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt [hunk 5]
```diff
@@ -83,3 +88,42 @@ internal fun ExternalIdResult.resolve(result: MethodChannel.Result) {
...
+        }.toSet()
+    }
+    val grantedPermissions = optJSONArray("grantedPermissions")?.let { outer ->
+        (0 until outer.length()).map { i ->
+            val inner = outer.getJSONArray(i)
+            (0 until inner.length()).map { j -> inner.getString(j) }.toSet()
+        }.toSet()
+    }
+    val malwareScanScope = optJSONObject("malwareScanScope")?.toMalwareScanScope()
+        ?: MalwareScanScope(ScopeType.SIDELOADED_ONLY, emptyList())
```
Comment (tompsota) [inline android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt]: Do not fallback to default args - it creates non-debuggable issues. Just throw exception

Comment (tompsota) [inline android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt]: The single-level and nested patterns each repeat a lot. The file already has a perfectly good generic helper for this — Utils.kt defines extractArray<T>() and processArray<T>() with reified types. 

Can you try to reuse those ?


Also, mapTo(mutableSetOf()) avoids building a List and then copying to a Set.

File: lib/src/models/suspicious_app_detection_config.dart [hunk 6]
```diff
@@ -0,0 +1,92 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'suspicious_app_detection_config.g.dart';
+
+/// The scope of apps to be scanned for malware.
+enum ScopeType {
+  /// Only sideloaded apps.
```
Comment (tompsota) [inline lib/src/models/suspicious_app_detection_config.dart:7]: Pls remove such comments, they don;t have a value

**Review by tompsota — CHANGES_REQUESTED** [review]
One more thing: There are no tests for SuspiciousAppDetectionConfig, MalwareScanScope, ScopeType, or ReasonMode. Every other model in lib/src/models/ has a corresponding test/src/models/*_test.dart covering construction, toJson, fromJson, and edge cases (see test/src/models/android_config_test.dart lines 45–91 for the established pattern).

Also missing: an AndroidConfig test exercising the suspiciousAppDetectionConfig field in both fromJson/toJson directions

Can you just generate some simple test cases?

If the requested changes here are relevant also for other repos, please update those as well.

File: android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt [hunk 7]
```diff
@@ -118,18 +118,12 @@ internal fun JSONObject.toSuspiciousAppDetectionConfig(): SuspiciousAppDetection
         }.toSet()
     }
     val malwareScanScope = optJSONObject("malwareScanScope")?.toMalwareScanScope()
+        ?: MalwareScanScope(ScopeType.SIDELOADED_ONLY, emptyList())
     val reasonModeStr = optString("reasonMode")
     val reasonMode = if (reasonModeStr.isNullOrEmpty()) {
```
Comment (tompsota) [inline android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt]: Can we set default in Dart to ReasonMode.HIGHEST_CONFIDENCE, make the argument required in kotlin and throw exception if missing?

File: android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt [hunk 8]
```diff
@@ -118,18 +118,12 @@ internal fun JSONObject.toSuspiciousAppDetectionConfig(): SuspiciousAppDetection
         }.toSet()
     }
     val malwareScanScope = optJSONObject("malwareScanScope")?.toMalwareScanScope()
+        ?: MalwareScanScope(ScopeType.SIDELOADED_ONLY, emptyList())
     val reasonModeStr = optString("reasonMode")
```
Comment (tompsota) [inline android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt]: org.json.JSONObject#optString(name) returns "" (empty string) for a missing key, not null. The next line correctly uses isNullOrEmpty() so it works, but the variable name reasonModeStr plus the isNullOrEmpty() check imply the value can be null — which it can't from this API. Slightly misleading and inconsistent with the optString("scanScope", "SIDELOADED_ONLY") pattern used 30 lines above, which uses the two-arg overload to provide a default.

**Review by tompsota — CHANGES_REQUESTED** [review]

File: example/lib/main.dart (hunk 1 above)
Comment (martinzigrai) [reply example/lib/main.dart]: Bumped to 8.0.0

File: CHANGELOG.md (hunk 2 above)
Comment (martinzigrai) [reply CHANGELOG.md]: Bumped to 8.0.0

File: lib/src/models/android_config.dart (hunk 3 above)
Comment (martinzigrai) [reply lib/src/models/android_config.dart]: Removed completely

File: lib/src/models/android_config.g.dart
Comment (martinzigrai) [reply lib/src/models/android_config.g.dart:1]: Manual edit 😅

File: android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt (hunk 4 above)
Comment (martinzigrai) [reply android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt]: Refactored

File: android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt (hunk 5 above)
Comment (martinzigrai) [reply android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt]: Refactored

Comment (martinzigrai) [reply android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt]: Refactored

File: lib/src/models/suspicious_app_detection_config.dart (hunk 6 above)
Comment (martinzigrai) [reply lib/src/models/suspicious_app_detection_config.dart:7]: CI will fail if I remove them

File: android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt (hunk 7 above)
Comment (martinzigrai) [reply android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt]: Refactored

File: android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt (hunk 8 above)
Comment (martinzigrai) [reply android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt]: Refactored

**Comment by martinzigrai** [comment]
Added tests for SuspiciousAppDetectionConfig, MalwareScanScope, ScopeType, ReasonMode plus AndroidConfig coverage for the suspiciousAppDetectionConfig field.

File: example/lib/main.dart [hunk 9]
```diff
@@ -38,12 +38,16 @@ Future<void> _initializeTalsec() async {
...
-      malwareConfig: MalwareConfig(
-        blacklistedPackageNames: ['com.aheaditec.freeraspExample'],
-        suspiciousPermissions: [
+      suspiciousAppDetectionConfig: const SuspiciousAppDetectionConfig(
+        packageNames: ['com.aheaditec.freeraspExample'],
+        requestedPermissions: [
           ['android.permission.CAMERA'],
           ['android.permission.READ_SMS', 'android.permission.READ_CONTACTS'],
         ],
+        malwareScanScope: MalwareScanScope(
```
Comment (tompsota) [inline example/lib/main.dart]: Upon discussion, rename this to: 
```suggestion
        scanScope: ScanScope(
```

**Review by tompsota — CHANGES_REQUESTED** [review]

**Review by tompsota — APPROVED** [review]
